### PR TITLE
[AC][SUPP-229] Missing information page

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/controllers/MissingInfoController.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import util.Logging
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class MissingInfoController @Inject() (
+    as:              Actions,
+    mcc:             MessagesControllerComponents,
+    missingInfoPage: views.html.MissingInformation
```
Comment (barnesjake) [inline app/controllers/MissingInfoController.scala:30]: Should this go into views and then use `views.missingInfoPage`. I'm not overly fussed if it doesn't but I think that's the pattern we have atm

Comment (barnesjake) [reply app/controllers/MissingInfoController.scala:30]: By views I mean the Views class we inject

File: app/controllers/PaymentScheduleController.scala [hunk 2]
```diff
@@ -45,25 +44,29 @@ class PaymentScheduleController @Inject() (
...
+        finalStateCheck(j, Ok(paymentSchedulePage(upfrontPaymentAnswers, paymentDay, j.selectedPaymentPlan)))
     }
   }
 
   val checkPaymentScheduleSubmit: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
     request.journey match {
-      case j: Journey.BeforeSelectedPaymentPlan =>
-        logErrorAndRouteToDefaultPageF(j)
+      case _: Journey.BeforeSelectedPaymentPlan =>
+        JourneyLogger.warn("Not enough information in session to show the check payment schedule page. " +
```
Comment (barnesjake) [inline app/controllers/PaymentScheduleController.scala:67]: Could we extract this warn into a def in JourneyLogger and just pass in the page as a string maybe, saves having the same thing over and over

File: app/controllers/MissingInfoController.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/controllers/MissingInfoController.scala:30]: 👍 I'll do that now

File: app/controllers/PaymentScheduleController.scala (hunk 2 above)
Comment (AndyHWChung) [reply app/controllers/PaymentScheduleController.scala]: just pushed up a change for this - it's quite specific so I wasn't sure if `JourneyLogger` was the best place to factor it out. I stuck something in the companion object of `MissingInfoController`. What do you think?

File: app/controllers/MissingInfoController.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/controllers/MissingInfoController.scala]: just changed it in [this commit ](https://github.com/hmrc/essttp-frontend/pull/143/commits/85d367c7ae53ca5b0985bd793ceaf18bb235bad9)

File: app/controllers/PaymentScheduleController.scala (hunk 2 above)
Comment (barnesjake) [reply app/controllers/PaymentScheduleController.scala]: Yeah that sounds sensible

**Review by barnesjake — APPROVED** [review]

**Pull request merged**